test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -603,4 +610,146 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
+				testing.MakeWorkload("four", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+			for _, w := range newWorkloads {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+				time.Sleep(time.Second)
+			}
+
+			ginkgo.By("Cut off the head of pending workloads when the number of pending workloads exceeds MaxCount")
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
+				return updatedCq.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{
+				Head: []kueue.ClusterQueuePendingWorkload{
+					{
+						Name:      "one",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "two",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "three",
+						Namespace: ns.Name,
+					},
+				},
+			}, ignoreLastChangeTime, cmpopts.SortSlices(func(i, j kueue.ClusterQueuePendingWorkload) bool {
+				return i.Name < j.Name
+			})))
+
+			ginkgo.By("Admitting workloads")
+			admissions := []*kueue.Admission{
+				testing.MakeAdmission(clusterQueue.Name).
+					Assignment(corev1.ResourceCPU, flavorOnDemand, "2").Obj(),
+				testing.MakeAdmission(clusterQueue.Name).
+					Assignment(corev1.ResourceCPU, flavorOnDemand, "3").Obj(),
+				testing.MakeAdmission(clusterQueue.Name).
+					Assignment(corev1.ResourceCPU, flavorOnDemand, "1").Obj(),
+				nil,
+			}
+			for i, w := range workloads {
+				gomega.Eventually(func() error {
+					var newWL kueue.Workload
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(w), &newWL)).To(gomega.Succeed())
+					if admissions[i] != nil {
+						return util.SetQuotaReservation(ctx, k8sClient, &newWL, admissions[i])
+					}
+					return k8sClient.Status().Update(ctx, &newWL)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			}
+
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCQ kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCQ)).To(gomega.Succeed())
+				return updatedCQ.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{
+				Head: []kueue.ClusterQueuePendingWorkload{
+					{
+						Name:      "three",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "four",
+						Namespace: ns.Name,
+					},
+				},
+			}, ignoreLastChangeTime, cmpopts.SortSlices(func(i, j kueue.ClusterQueuePendingWorkload) bool {
+				return i.Name < j.Name
+			})))
+
+			ginkgo.By("Finishing workload", func() {
+				util.FinishWorkloads(ctx, k8sClient, workloads...)
+				util.FinishWorkloads(ctx, k8sClient, newWorkloads...)
```

Comment:
Check that after finishing all workloads the queue of pending workloads is empty.

Reply:
check added